libraries/AP_Quicktune/AP_Quicktune.cpp

```diff
@@ -0,0 +1,730 @@
+        return Stage::P;
+    } else if (param == Param::RLL_I || param == Param::PIT_I || param == Param::YAW_I) {
+        return Stage::I;
+    } else if (param == Param::RLL_D || param == Param::PIT_D || param == Param::YAW_D) {
+        return Stage::D;
+    } else if (param == Param::RLL_SMAX || param == Param::PIT_SMAX || param == Param::YAW_SMAX) {
+        return Stage::SMAX;
+    } else if (param == Param::RLL_FLTT || param == Param::PIT_FLTT || param == Param::YAW_FLTT) {
+        return Stage::FLTT;
+    } else if (param == Param::RLL_FLTD || param == Param::PIT_FLTD || param == Param::YAW_FLTD) {
+        return Stage::FLTD;
+    } else if (param == Param::RLL_FLTE || param == Param::PIT_FLTE || param == Param::YAW_FLTE) {
+        return Stage::FLTE;
+    } else if (param == Param::RLL_FF || param == Param::PIT_FF || param == Param::YAW_FF) {
+        return Stage::FF;
+    } else {
+        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+        return Stage::END;
+    }
+}
+
+AP_Float *AP_Quicktune::get_param_pointer(AP_Quicktune::Param param)
+{
+    auto &attitude_control = *AC_AttitudeControl::get_singleton();
+    AC_PID* pid_ptr;
+
+    AxisName axis = get_axis(param);
+    switch (axis)
+    {
+        case AxisName::RLL:
+            pid_ptr = &attitude_control.get_rate_roll_pid();
+            break;
+        case AxisName::PIT:
+            pid_ptr =  &attitude_control.get_rate_pitch_pid();
+            break;
+        case AxisName::YAW:
+            pid_ptr =  &attitude_control.get_rate_yaw_pid();
+            break;
+        default:
+            INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+            return nullptr;
+    }
+    
+    Stage stage = get_stage(param);
+    switch (stage)
+    {
+        case Stage::P:
+            return &pid_ptr->kP();
+        case Stage::I:
+            return &pid_ptr->kI();
+        case Stage::D:
+            return &pid_ptr->kD();
+        case Stage::SMAX:
+            return &pid_ptr->slew_limit();
+        case Stage::FLTT:
+            return &pid_ptr->filt_T_hz();
+        case Stage::FLTD:
+            return &pid_ptr->filt_D_hz();
+        case Stage::FLTE:
+            return &pid_ptr->filt_E_hz();
+        case Stage::FF:
+            return &pid_ptr->ff();
+        default:
+            INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+            return nullptr;
+    }
+}
+
+float AP_Quicktune::get_param_value(AP_Quicktune::Param param)
+{
+    AP_Float *ptr = get_param_pointer(param);
+    if (ptr != nullptr) {
+        return ptr->get();
+    }
+    INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+    return 0.0;
+}
+
+void AP_Quicktune::set_param_value(AP_Quicktune::Param param, float value)
+{
+    AP_Float *ptr = get_param_pointer(param);
+    if (ptr != nullptr) {
+       ptr->set(value);
+       return;
+    }
+    INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+    return;
```

Comment:
```suggestion
```

Reply:
I removed the return keywords, but as noted below, I do think the internal error is kinda needed. Otherwise, if the pointer does end up being nullptr, the tuner will appear to be setting its parameters when it's actually not doing anything.

Reply:
Removed internal error as well.